### PR TITLE
Fix bug in CCG calculations for measures querying practice_statistics

### LINKED ISSUE #1520: Bug in CCG calculations for measures querying practice_statistics
There is a bug in our measure calculation code that affects measures where the value of a numerator or denominator for a practice can be null.

This affects the CCG ratios of measures that query the practice_statistics table, which is around half of our measures.

~specifically~

* ~ktt9_antibiotics~
* ~ktt9_cephalosporins_star~
* ~tamoxifen~

[Edit: I had originally thought that only three measures were affected..]

The problem is that if the denominator for a practice is null, the **numerator** for the corresponding measure value get set to zero (as well as the denominator, which is expected).  This doesn't affect the ratio for the practice measure, since if the denominator is zero we set the ratio to NULL.

However, it does affect the ratio for the CCG measure, because this is computed as the sum of all practice numerators over the sum of all practice denominators.  The sum of all practice denominators is never zero in practice so an incorrect sum of all practice numerators will give an incorrect ratio.

The denominator for a practice can be null when the denominator is selected from the practice_statistics table, because we don't have practice statistics for closed or dormant practices.

For example:

```
SELECT practice, month, SUM(items) AS numerator
FROM hscic.normalised_prescribing_standard
WHERE practice = 'Y02514' AND month = TIMESTAMP('2018-06-01') AND bnf_code LIKE '0501%'
GROUP BY practice, month
```

practice | month | numerator
-- | -- | --
Y02514 | 2018-06-01 00:00:00 UTC | 594

But:

```
SELECT practice_id, month, numerator, denominator, calc_value
FROM measures.practice_data_ktt9_antibiotics
WHERE practice_id = 'Y02514' AND month = DATE('2018-06-01')
```

practice_id | month | numerator | denominator | calc_value
-- | -- | -- | -- | --
Y02514 | 2018-06-01 | 0 | 0.0 | null

### PR DESCRIPTION
**Pull request by inglesp**
Fixes #1520.

### CODE REVIEW

File: openprescribing/frontend/tests/commands/test_import_measures_new.py [hunk 1]
```diff
@@ -190,15 +215,246 @@ def test_import_measures(self):
...
+
+                    prescribing_rows.append(row)
+
+        # In production, normalised_prescribing_standard is actually a view,
+        # but for the tests it's much easier to set it up as a normal table.
+        table = Client('hscic').get_or_create_table(
+            'normalised_prescribing_standard', schemas.PRESCRIBING_SCHEMA
+        )
+
+        # Upload prescribing_rows to normalised_prescribing_standard, and
```
Comment (sebbacon) [inline openprescribing/frontend/tests/commands/test_import_measures_new.py]: This is copy-and-paste from earlier, could be a function?

File: openprescribing/frontend/tests/commands/test_import_measures_new.py [hunk 2]
```diff
@@ -190,15 +215,246 @@ def test_import_measures(self):
...
+            'practice_id',
+            'total_list_size',
+        ]
+        practice_statistics = pd.DataFrame(columns=columns)
+
+        for practice in Practice.objects.all():
+            for month in [7, 8]:
+                timestamp = '2018-0{}-01 00:00:00 UTC'.format(month)
+
+                if month == 8 and practice.code == 'P00000':
```
Comment (sebbacon) [inline openprescribing/frontend/tests/commands/test_import_measures_new.py]: Comment explaining what's going on here?  Or perhaps an assertion like you do in the other test (`seen_practice_with_no_list_size = True` + `assert seen_practice_with_no_list_size`)?

File: openprescribing/frontend/tests/commands/test_import_measures_new.py [hunk 3]
```diff
@@ -190,15 +215,246 @@ def test_import_measures(self):
...
+                        'practice_id': practice.code,
+                        'ccg_id': practice.ccg_id,
+                        'stp_id': practice.ccg.stp_id,
+                        'regional_team_id': practice.ccg.regional_team_id,
+                        'total_list_size': total_list_size,
+                    },
+                    ignore_index=True,
+                )
+
+        # Upload practice_statistics_rows to BigQuery.
```
Comment (sebbacon) [inline openprescribing/frontend/tests/commands/test_import_measures_new.py]: Possibly a candidate for including in refactoring I mentioned above?

File: openprescribing/frontend/tests/commands/test_import_measures_new.py [hunk 4]
```diff
@@ -190,15 +215,246 @@ def test_import_measures(self):
...
+        )
+        self.validate_practice_statistics_measure_global(mg, practices, 'practice')
+        mvs = MeasureValue.objects.filter(
+            month=month,
+            practice_id__isnull=False,
+            pct_id__isnull=False,
+            stp_id__isnull=False,
+            regional_team_id__isnull=False,
+        )
+        self.assertEqual(mvs.count(), 625)
```
Comment (sebbacon) [inline openprescribing/frontend/tests/commands/test_import_measures_new.py]: What does this number relate to - number of practices presumably? Could it be a variable? (e.g. `Practice.objects.count()`)

File: openprescribing/frontend/tests/commands/test_import_measures_new.py [hunk 5]
```diff
@@ -208,15 +464,15 @@ def test_import_measures(self):
         )
         self.assertEqual(mvs.count(), 125)
```
Comment (sebbacon) [inline openprescribing/frontend/tests/commands/test_import_measures_new.py]: Number of regional teams?

File: openprescribing/frontend/tests/commands/test_import_measures_new.py [hunk 6]
```diff
@@ -226,15 +482,15 @@ def test_import_measures(self):
         )
         self.assertEqual(mvs.count(), 25)
```
Comment (sebbacon) [inline openprescribing/frontend/tests/commands/test_import_measures_new.py]: Number of STPs?

File: openprescribing/frontend/tests/commands/test_import_measures_new.py [hunk 7]
```diff
@@ -244,9 +500,13 @@ def test_import_measures(self):
         )
         self.assertEqual(mvs.count(), 5)
```
Comment (sebbacon) [inline openprescribing/frontend/tests/commands/test_import_measures_new.py]: Number of regions?

File: openprescribing/frontend/tests/commands/test_import_measures_new.py (hunk 4 above)
Comment (inglesp) [reply openprescribing/frontend/tests/commands/test_import_measures_new.py]: Good point!

File: openprescribing/frontend/tests/commands/test_import_measures_new.py (hunk 1 above)
Comment (inglesp) [reply openprescribing/frontend/tests/commands/test_import_measures_new.py]: Good idea.

File: openprescribing/frontend/tests/commands/test_import_measures_new.py (hunk 2 above)
Comment (inglesp) [reply openprescribing/frontend/tests/commands/test_import_measures_new.py]: Good idea.

**Comment by inglesp** [comment]
> why is there both a `test_import_measures` and a `test_import_measures_new`?

`test_import_measures_new` tests most of what's in `test_import_measures`, but more comprehensively, so most of `test_import_measures` can be dropped.  But I'm not sure there's much point doing this now, as `import_measures` will be going away soon.

**Pull request merged**